Packs/CommonScripts/Scripts/IsIPPrivate/IsIPPrivate.py

```diff
@@ -0,0 +1,116 @@
+                return tag
+    return None
+
+
+def create_markdown_table(ip_list):
+    ip_list = sorted(ip_list, key=lambda x: not x['Private'])
+    markdown_string = "| Address | Private | Tag |\n| --- | --- | --- |\n"
+    for ip in ip_list:
+        markdown_string += f"| {ip['Address']} | {ip['Private']} | {ip['Tag']}|\n"
+    return markdown_string
+
+
+ranges_list_name = demisto.args().get("PrivateIPsListName", "PrivateIPs")
+ip_addresses_to_check = argToList(demisto.args().get("IPAddresses", None))
+
+# Get the list of private IP ranges from the XSOAR list:
+private_ranges = demisto.executeCommand("getList", {"listName": ranges_list_name})[0]['Contents']
+if "Item not found" in private_ranges:
+    return_error(f"The list name {ranges_list_name} does not exist.")
+
+# Split ranges from XSOAR list to be a list:
+if private_ranges:
+    try:
+        private_ranges = private_ranges.split("\n")
+    except Exception as ex:
+        return_error(
+            "Could not parse the private ranges list. Please make sure that the list contains ranges written in CIDR notation, separated by new lines.")
+else:
+    private_ranges = ["10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"]  # No ranges in list, use default ranges
+
+# Create list of IPs with private property and tag
+ip_list = [{"Address": ip, "Private": check_ip_internal(ip, private_ranges), "Tag": get_ip_tag(ip, private_ranges)} for
+           ip in ip_addresses_to_check]
+
+# Create entry context and human-readable results
+entry_context = {"IP(val.Address == obj.Address)": ip_list}
+md_table = create_markdown_table(ip_list)
+entry_to_return = {
+    "Type": entryTypes['note'],
+    "Contents": ip_list,
+    "ContentsFormat": "text",
+    "HumanReadable": md_table,
+    "EntryContext": entry_context,
+    "Tags": ['IP_Private_Ranges']
+}
+
+for ip in ip_list:
+    args_exists_check = {
+        "indicator": ip.get("Address")
+    }
+
+    args_create_new_indicator = {
+        "value": ip.get("Address"),
+        "internal": ip.get("Private")
+    }
+
+    args_set_existing_indicator = {
+        "value": ip.get("Address"),
+        "internal": ip.get("Private")
+    }
+
+    args_append_tag = {
+        "field": "tags",
+        "indicatorsValues": ip.get("Address"),
+        "fieldValue": ip.get("Tag", "")
+    }
+
+    # Check if indicator exists:
+    is_exist_res = demisto.executeCommand("CheckIndicatorValue", args_exists_check)
+    indicator_exists = is_exist_res[0].get("Contents", {})[0].get("Exists")
+
+    # If indicator doesn't exist, create it:
+    if not indicator_exists:
+        demisto.executeCommand("createNewIndicator", args_create_new_indicator)
+        while not indicator_exists:  # Looping because it takes time for the indicator to be created
+            is_exist_res = demisto.executeCommand("CheckIndicatorValue", args_exists_check)
+            indicator_exists = is_exist_res[0].get("Contents", {})[0].get("Exists")
+            sleep(POLLING_TIME)
```

Comment:
No need for sleep, if indicator exists append tag, if not create with the tag.

Reply:
OK. I also changed it to "if indicator_exists" instead of "if not". It was unnecessarily confusing to check if NOT and then else.